### ISSUE TITLE
alpine musl build error

### ISSUE THREAD

**Issue by calvin2021y** [eBay/Jungle#143]
```sh
In file included from jungle/src/logger.cc:24:

jungle/src/backtrace.h:36:10: fatal error: 'execinfo.h' file not found
#include <execinfo.h>
         ^~~~~~~~~~~~
```

**Comment by greensky00** [comment]
@calvin2021y 
Sorry for the late reply. Actually `backtrace.h` was intended to work on either x86 Linux or Mac, so it should be disabled on platforms that do not support it (`execinfo.h`). Let me add a compiler flag to manually disable it, and push the PR shortly.

**Comment by calvin2021y** [comment]
alpine and musl do support backtrace and unwind, but some header is diff from glibc.

**Comment by greensky00** [comment]
Please check if https://github.com/eBay/Jungle/pull/145 and adding `-DLOGGER_NO_BACKTRACE=1` flag works for you.

**Comment by calvin2021y** [comment]
Thanks for the solution.

alpine is fixed for x86 and aarch64.

macOS show this error:

```sh
jungle/src/backtrace.h:330:24: error: use of undeclared identifier 'abi'
            char *cc = abi::__cxa_demangle(func_mangled.c_str(), 0, 0, &status);

```

**Comment by greensky00** [comment]
Now it should work on Mac with the flag, please check it again.

**Comment by calvin2021y** [comment]
Thanks for the update. some how I can not apply https://github.com/eBay/Jungle/pull/145.patch

check from the code it should be fixed. I will update and test late after you merge.

**Comment by calvin2021y** [comment]
one question about `Log store mode`:

it only can append data into store with a bigger SN. (can be discontinuous)

it can not delete by SN number, but can truncation by SN number ?

I also want to know if there will be supported with IO_uring async DISK io instead thread.

**Comment by greensky00** [comment]
> it can not delete by SN number, but can truncation by SN number ?

Right, it is not possible to delete a sequence number in the middle; you can only truncate it.

And currently there is no support for AsyncIO.
